controllers/eventbus/installer/nats.go

```diff
@@ -384,14 +384,17 @@ func (i *natsInstaller) createStatefulSet(ctx context.Context, serviceName, conf
 		if ss.Annotations != nil && ss.Annotations[common.AnnotationResourceSpecHash] == expectedSs.Annotations[common.AnnotationResourceSpecHash] {
 			return nil
 		}
-		// Delete the existing one to recreate it
-		err := i.client.Delete(ctx, ss)
-		if err != nil {
-			i.eventBus.Status.MarkDeployFailed("DeleteOldStatefulSetFailed", "Failed to delete a statefulset")
-			log.Errorw("error deleting a statefulset", zap.Error(err))
+		ss.Spec = expectedSs.Spec
```

Comment:
Updating existing statefulset instead of deleting and recreating to get better upgrade experience.